### PR TITLE
Prettify invoice details

### LINKED ISSUE #872: Add transaction details to transactions in wallet history
Upon clicking a transaction one should see some details to the transaction. There are different types of transactions for which a details view is required.

- [ ] On chain transaction (potentially add a link to mempool)
- [ ] Off-chain transaction (payment or received)
  - [ ] Invoice description (moved from #886)
- [ ] Position transaction (open and close)
- [ ] JIT Channel opening fee (potentially add a link to the funding transaction on mempool)
- [ ] Order matching fee

Figma mock: 

https://www.figma.com/file/Sr3SfZ46sZstKjSwftmTtc/10101-v1-(incl-bitcoin.design)?type=design&node-id=7375%3A45885&mode=design&t=tFOk1Qr2JvlNIaBY-1

E.g. like this

https://github.com/get10101/10101/assets/224613/1e989c69-6ba3-4c32-9117-f7f2ba850024

### PR DESCRIPTION
**Pull request by Restioson**
Fixes #872 by prettifying the invoice details modeled in #1111 and #1213 

To do:
- [x] Show failed payments better
- [x] Show expired invoices better
- [x] Show pending invoices better
- [x] Add mempool link

### CODE REVIEW

**Comment by Restioson** [comment]
<details>
<summary>Lightning payment (funding)</summary>

image

</details>

<details>
<summary>Channel opening fee</summary>

image

</details>


<details>
<summary>Opening position</summary>

image

</details>

<details>
<summary>Matching fee</summary>

image

</details>


<details>
<summary>Closed position</summary>

image

</details>

**Comment by bonomat** [comment]
Looks nice, but.. .would it be possible to use this dialog which comes up from the bottom as shown in the design [here](https://github.com/get10101/10101/issues/872#issue-1782067397)?

**Comment by Restioson** [comment]
> Looks nice, but.. .would it be possible to use this dialog which comes up from the bottom as shown in the design https://github.com/get10101/10101/issues/872#issue-1782067397?

Definitely should be possible, I can just swap out the container :)

**Comment by Restioson** [comment]
@bonomat: I've updated the screenshots

File: mobile/lib/common/middle_ellipsised_text.dart
Comment (Restioson) [inline mobile/lib/common/middle_ellipsised_text.dart:1]: This took me quite a while and I am still not quite happy with it, but it's good enough :sweat_smile:

**Comment by holzeis** [comment]
Nit: I don't think we need to copy the when 😅

Also the fonts seem pretty small can you make it bigger? And maybe have the content aligned to the right size with a bit space in between label and content? 😬

**Comment by bonomat** [comment]
> @bonomat: I've updated the screenshots

I think it looks better but I don't love it yet. I'm happy to play around with it if you don't have patience anymore.

**Comment by bonomat** [comment]
> @bonomat: I've updated the screenshots

I think it looks better but I don't love it yet. I'm happy to play around with it if you don't have patience anymore.

**Comment by Restioson** [comment]
@bonomat: sure, that's okay! If you have a clear vision for how it should look then go ahead :)

**Comment by Restioson** [comment]
Note to self:
> I think there is a lot of white in between and the text looks a bit small... but this is really just nitpicking [..]

Will try fix these for merge

**Comment by Restioson** [comment]
New look:
image

**Review by bonomat — APPROVED** [review]
Looks nice, thanks for going over it once more.

**Pull request merged**